config-model/src/main/java/com/yahoo/vespa/model/container/http/AccessControl.java

```diff
@@ -154,14 +153,17 @@ private void removeDuplicateBindingsFromChain(Http http, ComponentId chainId) {
         duplicateBindings.forEach(http.getBindings()::remove);
     }
 
-    private static boolean effectivelyDuplicateOf(BindingPattern accessControlBinding, BindingPattern other) {
-        return accessControlBinding.equals(other)
-                || (accessControlBinding.path().equals(other.path()) && other.matchesAnyPort());
+    private static boolean effectivelyDuplicateOf(FilterBinding accessControlBinding, FilterBinding other) {
+        if (accessControlBinding.chainId().equals(other.chainId())) return false; // Same filter chain
+        if (other.type() == FilterBinding.Type.RESPONSE) return false;
```

Comment:
How about comparing the types, instead of assuming this is a request type?

Reply:
The first parameter is assumed to be an access control binding, meaning it cannot be a response filter binding.

Reply:
Well, it's a private in the `AccessControl` class, so I guess that's OK.